packages/marshal/src/marshal.js

```diff
@@ -377,6 +374,20 @@ function assertRemotable(val) {
   }
 }
 
+/** @type {MarshalGetInterfaceOf} */
+const getInterfaceOf = val => {
+  if (typeof val !== 'object' || val === null) {
+    return undefined;
+  }
+  if (val[PASS_STYLE] !== REMOTE_STYLE) {
+    return undefined;
+  }
+  assertRemotable(val);
```

Comment:
I can appreciate the desire to be crisp, but I would prefer that `getInterfaceOf` never throw.  It is used in the REPL to decide how to render objects, and if it throws, that makes for a return value that cannot be rendered.

Reply:
Hi @michaelfig please take a look at my latest commit
https://github.com/Agoric/agoric-sdk/pull/2460/commits/ef6c0d328f7c6b55ff2ee5536d220b6a75b30b8e
which contains the comment
```js
// Below we have a series of paired predicate functions and assertion
// functions. The semantics of the assertion function is just to assert that
// the corresponding predicate function would have returned true. But it
// reproduces the internal tests so failures can give a better error message.
// Please maintain these pairs so their internal logic means the same thing,
// and so that they read as parallel to each other as possible.
```

I dislike the redundancy and maintenance hazard. I've done the same thing elsewhere though not as severely. I would rather have a decent pattern for expressing the logic once in a reusable manner, and parameterizing it so that it can act either like a predicate or like an *informative* assertion. But I couldn't come up with anything. Suggestions?

Reply:
> I dislike the redundancy and maintenance hazard. I've done the same thing elsewhere though not as severely. I would rather have a decent pattern for expressing the logic once in a reusable manner, and parameterizing it so that it can act either like a predicate or like an _informative_ assertion. But I couldn't come up with anything. Suggestions?

Something like:

```js
function checkPredicates(specimen, check = x => x) {
  if (!check(specimen.a === 'foo', X`a must be foo`)) {
    return false;
  }
  // ...
  return true;
}

// Run as an assertion.
checkPredicates(specimen, assert);

// Run as a check.
if (checkPredicates(specimen)) {
  // Do something.
}
```

Reply:
Good! I'll do that.

Reply:
A problem with literally the pattern above is that `assert` doesn't return `true` in the success case. Looks like I'll need to wrap `assert` too :(

Reply:
I've implemented that pattern, and wrapped `assert`.